### PR TITLE
event schema

### PR DESCRIPTION
**Pull request by pbugni**
Extending the README to include suggested details for how to format `event`s to make for consistent queries, and track all necessary data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,65 @@
...
+JSON data, intended to capture the intent of the event, such as "new consent
+signed" or "search for <...> found 0 matches".  Message generally captures
+the specific context from the code of the event being tracked, with all other
+details collected by a routine that can collect and populate the other top
+level attributes as specified below.
+
+The following should be common to all events on a given system:
+```json
+{
+    "version": "1", // the event schema version, not the application version
```
Comment (mcjustin) [inline README.md]: I'd prefer to see this labelled as such, eg "events_version"

File: README.md [hunk 2]
```diff
@@ -1,8 +1,65 @@
...
+    "name": "", // Application code package name, often built in to the logging system and difficult to manipulate
+    "level": "INFO", // Built in to the logging package, options also include DEBUG, WARN, ERROR
+```
+
+System identifiers to uniquely specify the source of the event:
+```json
+    "clinical-site": "", // unique name when appropriate to define jurisdiction, institution or clinic, such as "UW Harborview",
+    "deployment": "", // one of ["dev", "test", "demo", "stage", "prod"]
+    "system-type": "", // such as "remote" or "kiosk", if applicable
+    "system-name": "", // system identifier URL
```
Comment (mcjustin) [inline README.md]: I'd prefer "system-url" or "url" to disambiguate from "name" above

File: README.md [hunk 3]
```diff
@@ -1,8 +1,65 @@
...
+the specific context from the code of the event being tracked, with all other
+details collected by a routine that can collect and populate the other top
+level attributes as specified below.
+
+The following should be common to all events on a given system:
+```json
+{
+    "version": "1", // the event schema version, not the application version
+    "asctime": "", // ISO-8601 format including time-zone offset, preferably in UTC
+    "name": "", // Application code package name, often built in to the logging system and difficult to manipulate
```
Comment (mcjustin) [inline README.md:29]: prefer "application-name" or "app-name"

File: docs/cosri_v0.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    "levelname": "INFO"
+}
+```
+
+# FEMR (patientsearch) initiated logout
+```json
+{
+    "name": "patientsearch",
+    "tags": ["logout"],
+    "user": {
```
Comment (mcjustin) [inline docs/cosri_v0.md:55]: Looks like "user" was dropped from the main spec - I'd like to see it there...

**Review by mcjustin — COMMENTED** [review]
Many thanks @pbugni , a few minor suggestions

File: README.md (hunk 1 above)
Comment (pbugni) [reply README.md]: good idea!

File: README.md (hunk 2 above)
Comment (pbugni) [reply README.md]: another good idea!

File: README.md (hunk 3 above)
Comment (pbugni) [reply README.md:29]: built in to language logging libraries - difficult to change

File: docs/cosri_v0.md (hunk 4 above)
Comment (pbugni) [reply docs/cosri_v0.md:55]: ah!  thanks, oversight.

File: README.md [hunk 5]
```diff
@@ -1,8 +1,72 @@
...
+details collected by a routine that can collect and populate the other top
+level attributes as specified below.  
+
+Since we anticipate that events will undergo automated processing and there is only a single 'message' attribute, it seems likely that a message will have several json attributes, rather than be a simple string.  But, either is legal.
+
+The following should be common to all events on a given system:
+```json
+{
+    "event_version": "1", // the event schema version
+    "asctime": "", // ISO-8601 format including time-zone offset, preferably in UTC
```
Comment (ivan-c) [inline README.md]: I think just including a timezone _should_ be enough. The user's TZ may be a useful piece of information to collect
```suggestion
    "asctime": "", // ISO-8601 format including time-zone offset
```

File: docs/cosri_v1.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    "clinical-site": "Demo",
+    "deployment": "demo",
+    "system-name": "http://pdmp.cosri-demo.cirg.washington.edu/",
+    "message": "PDMP found match",
+    "patient": {
+        "subject.id": "2",
+	"subject:Patient.birthdate": "eq1974-09-01",
+	"subject:Patient.name.given": "Harry",
+	"subject:Patient.name.family": "Osborn"
+    },
```
Comment (daniellrgn) [inline docs/cosri_v1.md:38]: Should this key be "subject" as used elsewhere instead of "patient"?

**Review by ivan-c — APPROVED** [review]

File: docs/cosri_v1.md (hunk 6 above)
Comment (pbugni) [reply docs/cosri_v1.md:38]: that example is coming straight from the FHIR `Patient` resource, but I like your thinking - will see if there are other places we can work in `subject` more consistently.

**Comment by pbugni** [comment]
Probably not perfect, but merging to move the conversation and ability to link to develop branch forward.

**Pull request merged**